src/main/java/org/reactome/release/Resource.java

```diff
@@ -0,0 +1,166 @@
+		Map<String, String> mandatoryAttributesMap = new HashMap<>();
+
+		mandatoryAttributesMap.put("Release Step", getReleaseStep());
+		mandatoryAttributesMap.put("Main Program", getMainProgramName());
+		mandatoryAttributesMap.put("Dependency in Source Code", getSourceCodeDependency());
+		mandatoryAttributesMap.put("Resource", getResourceName());
+		mandatoryAttributesMap.put("Resource Description", getResourceDescription());
+		mandatoryAttributesMap.put("Resource Type", getResourceType().toString());
+		mandatoryAttributesMap.put("Resource URL", getResourceURL().toString());
+
+		for (Entry<String, String> mandatoryAttributeEntry: mandatoryAttributesMap.entrySet()) {
+			String mandatoryAttributeName = mandatoryAttributeEntry.getKey();
+			String mandatoryAttributeValue = mandatoryAttributeEntry.getValue();
+
+			if (mandatoryAttributeValue == null || mandatoryAttributeValue.isEmpty()) {
+				throw new IllegalStateException(
+					mandatoryAttributeName + " is empty or does not exist for " + originalRecord
+				);
+			}
+		}
+	}
+
+	public String getReleaseStep() {
+		return this.resourceAsJson.get("Release Step").getAsString();
+	}
+
+	public String getMainProgramName() {
+		return this.resourceAsJson.get("Main Program").getAsString();
+	}
+
+	public String getSourceCodeDependency() {
+		return this.resourceAsJson.get("Dependency in Source Code").getAsString();
+	}
+
+	public String getResourceName() {
+		return this.resourceAsJson.get("Resource").getAsString();
+	}
+
+	public String getResourceDescription() {
+		return this.resourceAsJson.get("Resource Description").getAsString();
+	}
+
+	public ResourceType getResourceType() {
+		String resourceType = this.resourceAsJson.get("Resource Type").getAsString().toUpperCase().replace(" ", "_");
+		return ResourceType.valueOf(resourceType);
+	}
+
+	public JsonObject getResourceAsJsonObject() {
+		return this.resourceAsJson;
+	}
+
+//	public String getHeaderAsTSVString() {
+//		return String.join("\t", getHeaderNameToValueMap().keySet());
+//	}
+//
+//	public String getValuesAsTSVString() {
+//		return String.join("\t", getHeaderNameToValueMap().values());
+//	}
+
+	public URL getResourceURL() {
+		String resourceURL = this.resourceAsJson.get("Resource URL").getAsString();
+		try {
+			return new URL(resourceURL);
+		} catch (MalformedURLException e) {
+			String errorMessage = "Unable to create URL for " + resourceURL;
+
+			logger.error(errorMessage, e);
+			throw new RuntimeException(errorMessage, e);
+		}
+	}
+
+	public String getErrorResponseText() {
+		JsonElement errorResponseText = this.resourceAsJson.get("Error Response Text");
+		return errorResponseText != null ? errorResponseText.getAsString() : "";
+	}
+
+	public String getExpectedResponseText() {
+		JsonElement expectedResponseText = this.resourceAsJson.get("Expected Response Text");
+		return expectedResponseText != null ? expectedResponseText.getAsString() : "";
+	}
+
+	public long getExpectedFileSizeInBytes() {
```

Comment:
The name of this suggests there is also an "actual" file size, somewhere. How many external resources are that static?

Reply:
This represents the size of the file the last time it was obtained.  It's just a measure to ensure that hasn't been a radical drop in file size.

Reply:
So it updates the CSV (or JSON) with the filesize each time it downloads a file? If a filesize is "too low", will it still record that in the CSV/JSON or not? Does a filesize that's too low cause an exception and termination of the program, or just log a warning?

Reply:
Never mind, I found the part that deals with filesizes...

Reply:
I haven't implemented the update to the file size yet, but yes, that's the plan.  The expected file size will be updated if it's found to be okay but not if the program finds a problem.  I also want to implement something if the file grows by an unreasonably large amount, but I'm still thinking about the details for that in terms of what a threshold would/shoud be.

Reply:
So right now... if a file grows over several releases, is it possible that eventually it will be "too big" and trigger an error?

Reply:
Right now, no matter how large the file gets it won't trigger any error but I do want to add logic that will do this soon.  I'm just thinking about how to do it and when we would consider a file is too big compared to what it used to be.

Reply:
Created issue #2